homework/unique_ptr/UniquePtr.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <iostream>
+#include <stdexcept>
+
+class DereferenceNullPtrError : public std::logic_error {
+public:
+    DereferenceNullPtrError()
+        : std::logic_error("Error - nullptr dereference") {}
+};
+
+template <typename T>
+class UniquePtr {
+public:
+    UniquePtr<T>() = default;
+    UniquePtr<T>(T* ptr);
+    ~UniquePtr();
+    UniquePtr(const UniquePtr& otherPtr) = delete;
+    UniquePtr(UniquePtr&& otherPtr);
+    UniquePtr& operator=(const UniquePtr& otherPtr) = delete;
+    UniquePtr& operator=(UniquePtr&& otherPtr);
+    T& operator*() const;
+    T* operator->() const;
+    T* get() const;
+    T* release();
+    void reset(T* newPtr);
```

Comment:
```suggestion
    void reset(T* newPtr = nullptr);
```
default argument would be useful :)